Core/include/Acts/Utilities/AngleHelpers.hpp

```diff
@@ -9,26 +9,120 @@
 #pragma once
 
 #include <cmath>
+#include <numbers>
 
 namespace Acts::AngleHelpers {
 
+template <typename Scalar>
+struct EtaThetaConversionTraits {};
+
+template <>
+struct EtaThetaConversionTraits<float> {
+  static constexpr float minTheta = 1e-6f;
+  static constexpr float maxTheta = std::numbers::pi_v<float> - 1e-6f;
+
+  static constexpr float minEta = -80.0f;
+  static constexpr float maxEta = 80.0f;
+};
+
+template <>
+struct EtaThetaConversionTraits<double> {
+  static constexpr double minTheta = 1e-12;
+  static constexpr double maxTheta = std::numbers::pi - 1e-12;
+
+  static constexpr double minEta = -700.0;
+  static constexpr double maxEta = 700.0;
+};
+
+template <typename Scalar>
+struct ClampedEtaThetaConversionTraits {
+  // computed from maxEta
+  static constexpr Scalar minTheta =
+      static_cast<Scalar>(3.6097027756908303447e-35L);
+  // computed from minEta
+  static constexpr Scalar maxTheta = static_cast<Scalar>(3.141592653589793);
```

Comment:
```suggestion
  static constexpr Scalar maxTheta = static_cast<Scalar>(3.1415926535897932385L);
```
sorry, I should have mentioned this one explicitly. This is for eta=-80.